### PR TITLE
Bring SecManager code under ENABLE_WHOAMI compile flag

### PR DESCRIPTION
**Pull request by PaviP98**

### CODE REVIEW

File: MaintenanceManager/MaintenanceManager.cpp
Comment (StefanBossbaly) [inline MaintenanceManager/MaintenanceManager.cpp:84]: nit: Add `/* defined(ENABLE_WHOAMI) */` at the `#endif`

File: MaintenanceManager/MaintenanceManager.cpp
Comment (StefanBossbaly) [inline MaintenanceManager/MaintenanceManager.cpp:247]: nit: Add `/* defined(ENABLE_WHOAMI) */` at the `#endif`

File: MaintenanceManager/MaintenanceManager.cpp
Comment (StefanBossbaly) [inline MaintenanceManager/MaintenanceManager.cpp:284]: nit: Add `/* defined(ENABLE_WHOAMI) */` at the `#endif`

File: MaintenanceManager/MaintenanceManager.cpp
Comment (StefanBossbaly) [inline MaintenanceManager/MaintenanceManager.cpp:491]: nit: Add `/* defined(ENABLE_WHOAMI) */` at the `#endif`

File: MaintenanceManager/MaintenanceManager.h
Comment (StefanBossbaly) [inline MaintenanceManager/MaintenanceManager.h:147]: nit: Add `/* defined(ENABLE_WHOAMI) */` at the `#endif`

File: MaintenanceManager/CHANGELOG.md
Comment (StefanBossbaly) [inline MaintenanceManager/CHANGELOG.md:26]: Do we need to bump the API version twice from `20` -> `22`? Both changes will take effect atomically so there will be no clients that use the `21` version of this API. Is it because they are different categories?

**Review by StefanBossbaly — COMMENTED** [review]
LGTM with a couple changes. We need to hold off until my WhoAmI changes get merged since it will cause a merge conflict.

**Review by anand-ky — APPROVED** [review]

**Pull request merged**
